From b12ea7fe7c05305a3e89922f6dd40a3a3bf97fa9 Mon Sep 17 00:00:00 2001
From: Leon Hielscher <Hielscher@fzi.de>
Date: Wed, 10 Oct 2018 13:08:37 +0200
Subject: [PATCH] RIOT: Use RIOT random_bytes function instead of randombytes

---
 snapshots/hacl-c/haclnacl.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/snapshots/hacl-c/haclnacl.c b/snapshots/hacl-c/haclnacl.c
index 42e9dc3..fcd629f 100644
--- a/snapshots/hacl-c/haclnacl.c
+++ b/snapshots/hacl-c/haclnacl.c
@@ -50,7 +50,7 @@
 #include "NaCl.h"
 
 
-extern void randombytes(uint8_t *bytes, uint64_t bytes_len);
+#include "random.h"
 
 
 /* HACL* Primitives and Constructions */
@@ -115,7 +115,7 @@ int crypto_auth_verify(const uint8_t *tag, const uint8_t *input, uint64_t input_
 
 
 int crypto_box_keypair(uint8_t *pk, uint8_t *sk){
-  randombytes(sk, 32);
+  random_bytes(sk, 32);
   uint8_t basepoint[32] = {9};
   curve25519_scalarmult(pk, sk, basepoint);
   return 0;
@@ -263,7 +263,7 @@ int crypto_sign_open(uint8_t *unsigned_msg, uint64_t *unsigned_msg_len, const ui
 }
 
 int crypto_sign_keypair(uint8_t pk[32], uint8_t sk[64]){
-  randombytes(sk, 32 * sizeof(uint8_t));
+  random_bytes(sk, 32 * sizeof(uint8_t));
   Hacl_Ed25519_secret_to_public(pk, sk);
   for (int i = 0; i < 32; i++) sk[32+i] = pk[i];
   return 0;
-- 
2.7.4

@s